module/VuFind/src/VuFind/View/Helper/Root/Csp.php

```diff
@@ -72,8 +72,12 @@ public function disablePolicy()
             as $field
         ) {
             if ($cspHeaders = $headers->get($field)) {
-                foreach ((array)$cspHeaders as $header) {
-                    $headers->removeHeader($header);
+                if ($cspHeaders instanceof \ArrayIterator) {
```

Comment:
Perhaps, but I'm now spooked about the possibility of a single header to be somehow iterable, so I wanted to be explicit about it.

Reply:
If you think the risk is minimal, I'm happy to revise.

Reply:
I see from the laminas-http documentation that `get()` is always supposed to return an `ArrayIterator` for multiple headers, so I think your check does make sense. I wonder if it would be more concise to do something like this, though:

<pre>
$headerArray = $cspHeaders instanceof \ArrayIterator
    ? $cspHeaders : [$cspHeaders];
foreach ($headerArray as $header) {
    $headers->removeHeader($header);
}
</pre>

This reduces the amount of indentation and eliminates some minor duplication of logic.

Reply:
Good idea, done. I also added a comment to clarify.

Reply:
Thanks! Looks good to merge now.